elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -71,18 +82,59 @@ public static void main(String[] args) throws IOException, ParseException {
             throw new MissingOptionException("Missing required option");
         }
         String configDir = cli.getOptionValue("configDir");
-        File file = new File(configDir);
-        String absoluteBasePath = file.getAbsolutePath();
-        log.info("Absolute Path for Model Configs Directory: " + absoluteBasePath);
 
-        if (!file.isDirectory()) {
+        DynamicConfigValidator dynamicConfigValidator = new DynamicConfigValidator();
+
+        if (exists(configDir)) {
+            //look in local file system
+            File configFile = new File(configDir);
+            String absoluteBasePath = configFile.getAbsolutePath();
+            log.info("Absolute Path for Model Configs Directory: " + absoluteBasePath);
+
+            dynamicConfigValidator.readAndValidateConfigs(absoluteBasePath);
+
+        } else {
+            // look in classpath
+            log.info("Looking at " + configDir + " in resources");
+            dynamicConfigValidator.readAndValidateClasspathConfigs(configDir);
+        }
+
+        log.info("Configs Validation Passed!");
+    }
+
+    /**
+     * Read and validate config files under resources.
+     * @throws IOException
+     */
+    public void readAndValidateClasspathConfigs(String filePath)
+            throws IOException {
+
+        this.setConfigDir(DynamicConfigHelpers.formatFilePath(formatClassPath(filePath)));
+        ClassLoader classLoader = DynamicConfigValidator.class.getClassLoader();
+
+        URL url = classLoader.getResource(this.configDir);
+        if (url == null) {
             throw new IllegalStateException("Model Configs Directory doesn't exists");
         }
 
-        DynamicConfigValidator dynamicConfigValidator = new DynamicConfigValidator();
-        dynamicConfigValidator.readAndValidateConfigs(absoluteBasePath);
+        FileSystem fs = null;
 
-        log.info("Configs Validation Passed!");
+        try {
+            URL jar = DynamicConfigValidator.class.getProtectionDomain().getCodeSource().getLocation();
```

Comment:
I still think there might be a simpler way to do this:
https://stackoverflow.com/questions/24078317/get-directory-from-classpath-getting-a-file-now